Rules/UseProcessBlockForPipelineCommand.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class UseProcessBlockForPipelineCommand : IScriptRule
+    {
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            IEnumerable<Ast> functionAsts = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);
+
+            foreach (FunctionDefinitionAst funcAst in functionAsts)

```

Comment:
Surprised this compiles without a typecast or a filter

Reply:
@rjmholt Any reason why it wouldn't?

Reply:
You're supposed to have an `IEnumerable<Ast>` which could contain any `Ast` type (for example, `ExpressionAst`), rather than just `FunctionDefinitionAst`s. So it seems here there is an implicit downcast, which could conceivably cause a runtime error -- seems a bit odd, and I could have sworn I've had to use an explicit cast in this exact case in the past.

Reply:
TBH I find the same usage all over the project, and I think I just robbed it from another rule. However, I have no issue changing it if necessary.

Reply:
If it compiles then no worries. Just surprised that it does